### PR TITLE
Document new node drain policies and clarify drain behavior

### PR DESCRIPTION
**Pull request by ejweber**
longhorn/longhorn#2238

- Mention new `Node Drain Policy` options in `v1.6.0` notes.
- Mention field deprectation in `v1.6.0` notes.
- Clarify what happens during a drain.
- Add a section comparing and contrasting different node drain policies (including the new ones).

### CODE REVIEW

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md [hunk 1]
```diff
@@ -12,91 +12,229 @@ This section describes how to handle planned node maintenance or upgrading Kuber
...
 
 ## Updating the Node OS or Container Runtime
 
 1. Cordon the node. Longhorn will automatically disable the node scheduling when a Kubernetes node is cordoned.
 
 1. Drain the node to move the workload to somewhere else.
 
-    You will need to use `--ignore-daemonsets` to drain the node.
-    The `--ignore-daemonsets` is needed because Longhorn deployed some daemonsets such as `Longhorn manager`, `Longhorn CSI plugin`, `engine image`.
+   You will need to use `--ignore-daemonsets` to drain the node. The `--ignore-daemonsets` is needed because Longhorn
```
Comment (james-munson) [inline content/docs/1.6.0/volumes-and-nodes/maintenance.md:23]: ~You will need to use~ 
It will be necessary to use ...

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md [hunk 2]
```diff
@@ -12,91 +12,229 @@ This section describes how to handle planned node maintenance or upgrading Kuber
...
-    The running replicas on the node will be stopped at this stage. They will be shown as `Failed`.
+   While the drain proceeds, engine processes on the node will be migrated with the workload pods to other nodes.
 
-   > **Note:**
-   > By default, if there is one last healthy replica for a volume on
-   > the node, Longhorn will prevent the node from completing the drain
-   > operation, to protect the last replica and prevent the disruption of the
-   > workload. You can control this behavior in the setting [Node Drain Policy](../../references/settings#node-drain-policy), or [evict
-   > the replica to other nodes before draining](../disks-or-nodes-eviction).
+   > **Note:** For volumes that are not attached through the CSI flow on the node (for example, manually attached using
```
Comment (james-munson) [inline content/docs/1.6.0/volumes-and-nodes/maintenance.md:28]: ~For~ Volumes that are not attached ... 
~they~ will not be automatically attached ...

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md [hunk 3]
```diff
@@ -12,91 +12,229 @@ This section describes how to handle planned node maintenance or upgrading Kuber
...
+   > UI), they will not be automatically attached to new nodes by Kubernetes during the draining. Therefore, Longhorn
+   > will prevent the node from completing the drain operation. User would need to handle detachment for these volumes
```
Comment (james-munson) [inline content/docs/1.6.0/volumes-and-nodes/maintenance.md:30]: ~User would need to handle detachment for~  The user will need to detach these volumes manually ...

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md [hunk 4]
```diff
@@ -12,91 +12,229 @@ This section describes how to handle planned node maintenance or upgrading Kuber
...
+   > Recommendations](#node-drain-policy-recommendations) for considerations when selecting a policy.
 
-   > **Note:** Normally you don't need to evict the replicas before the drain
-   > operation, as long as you have healthy replicas on other nodes. The replicas
-   > can be reused later, once the node back online and uncordoned.
+   After the drain is completed, there should be no engine or replica processes running on the node, as the
+   instance-manager pod that was running them will be stopped. Depending on the [Node Drain
+   Policy](#node-drain-policy-recommendations), replicas scheduled to the node will either appear as `Failed` or be
+   removed in favor of replacements. Workloads using Longhorn volumes will function as expected and enough replicas will
+   be running elsewhere to meet the requirements of the policy.
```
Comment (james-munson) [inline content/docs/1.6.0/volumes-and-nodes/maintenance.md:46]: ✔️

**Review by james-munson — DISMISSED** [review]
A couple of nits, but that's a really clear and comprehensive explanation.

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md (hunk 1 above)
Comment (ejweber) [reply content/docs/1.6.0/volumes-and-nodes/maintenance.md:23]: Fixing in an amend.

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md (hunk 2 above)
Comment (ejweber) [reply content/docs/1.6.0/volumes-and-nodes/maintenance.md:28]: Fixing in an amend.

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md (hunk 3 above)
Comment (ejweber) [reply content/docs/1.6.0/volumes-and-nodes/maintenance.md:30]: Fixing in an amend.

File: content/docs/1.6.0/volumes-and-nodes/maintenance.md [hunk 5]
```diff
@@ -12,91 +12,229 @@ This section describes how to handle planned node maintenance or upgrading Kuber
...
 
-   > **Note:**
-   > By default, if there is one last healthy replica for a volume on
-   > the node, Longhorn will prevent the node from completing the drain
-   > operation, to protect the last replica and prevent the disruption of the
-   > workload. You can control this behavior in the setting [Node Drain Policy](../../references/settings#node-drain-policy), or [evict
-   > the replica to other nodes before draining](../disks-or-nodes-eviction).
+   > **Note:** Volumes that are not attached through the CSI flow on the node (for example, manually attached using
+   > UI) will not be automatically attached to new nodes by Kubernetes during the draining. Therefore, Longhorn will
+   > prevent the node from completing the drain operation. The user will need to detach these volumes manually to
```
Comment (james-munson) [inline content/docs/1.6.0/volumes-and-nodes/maintenance.md:30]: Looks better, although this will almost immediately be outdated by the addition of another setting that will enable auto-detachment of manually attached volumes.  But that can be its own PR.

**Review by james-munson — APPROVED** [review]

**Review by innobead — APPROVED** [review]
LGTM.
